### DISCUSSION TITLE
Real gas phase

### DISCUSSION THREAD

**Discussion by corykinney** [ReactionMechanismGenerator/ReactionMechanismSimulator.jl#224] [Ideas]
Hello! Is there any interest in adding real gas phase support to RMS? I'd be interested in contributing, but I wanted to ask before starting anything to make sure it's acceptable and to ask for pointers on how to go about implementing it. Thanks.

**Comment by mjohnson541** [comment]
Yes, discussions we've had about this revolve around integrating Clapeyron.jl https://github.com/ClapeyronThermo/Clapeyron.jl into RMS (see https://github.com/ReactionMechanismGenerator/ReactionMechanismSimulator.jl/issues/157). Such an integration would allow use of arbitrary thermodynamic models in RMS

Generating a real gas phase using Clapeyron.jl might be a good first step towards that. Inside RMS I think first you would need to add an object type in Calculators/Thermo.jl that can hold the real gas information and then create a RealGas phase in Phase.jl that is mostly IdealGas except that the vectorized thermo (generated from the individual real gas thermo information for each species) calls Clapeyron.jl to compute the thermochemistry of each species. @pw0908 and @longemen3000 do you guys have any advice here? Is there a good example for real gas thermo in Clapeyron.jl?

**Comment by longemen3000** [comment]
With 1.9, this could be inside a Julia pkg extension.

**Comment by corykinney** [comment]
Does Clapeyron.jl account for activity coefficients in addition to thermodynamic state?

**Comment by longemen3000** [comment]
If you use an activity coefficient model, then activities will be considered (in the liquid phase, you have to choose another model for the gas phase). Models like PC-SAFT, or the multiparameter equations of state, are not defined in terms of activities, but nonetheless they still encode deviations from ideality in the liquid phase.
